nexus/language.ts

```diff
@@ -155,9 +76,72 @@ schema.extendType({
 schema.extendType({
   type: 'Mutation',
   definition(t) {
-    t.field('addLanguageLearning', addLanguageM2MMutation('LanguageLearning'))
-    t.field('addLanguageNative', addLanguageM2MMutation('LanguageNative'))
-    t.field('removeLanguageLearning', removeLanguageM2MMutation('LanguageLearning'))
-    t.field('removeLanguageNative', removeLanguageM2MMutation('LanguageNative'))
+    t.field('addLanguageRelation', {
+      type: 'LanguageRelation',
+      args: {
+        languageId: schema.intArg({ required: true }),
+        level: schema.arg({ type: 'LanguageLevel', required: true }),
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+
+        if (!userId) {
+          throw new Error('You must be logged in add language relations.')
+        }
+
+        const language = await ctx.db.language.findOne({
+          where: { id: args.languageId },
+        })
+
+        if (!language) {
+          throw new Error(`Unable to find language with id "${args.languageId}".`)
+        }
+
+        return ctx.db.languageRelation.create({
+          data: {
+            user: { connect: { id: userId } },
+            language: { connect: { id: args.languageId } },
+            level: args.level,
+          },
+        })
+      }
+    })
+    t.field('removeLanguageRelation', {
+      type: 'LanguageRelation',
+      args: {
+        languageId: schema.intArg({ required: true }),
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+
+        if (!userId) {
+          throw new Error('You must be logged in remove language relations.')
+        }
+
+        const relFilter = {
+          where: {
+            userId_languageId: {
+              languageId: args.languageId,
+              userId,
+            },
+          },
+        }
+
+        const relation = await ctx.db.languageRelation.findOne({
+          where: {
+            userId_languageId: {
+              languageId: args.languageId,
+              userId,
+            },
+          },
+        })
```

Comment:
Would this work?

```suggestion
        const relation = await ctx.db.languageRelation.findOne(relFilter)
```

Reply:
Yep! I think after the refactor I just thought that it made sense to simply do it inside the `where` since `relFilter` was only being used once now

Reply:
Yes and no! bahaha yes on a basic level to use the variable instead, but it looks like this is now an invalid `languageRelation.findOne()` invocation o(>.<;)o

Reply:
Weird though because digging around the `index.d.ts` file I still see `userId_languageId` as it was, so maybe something else is causing the error